src/Yi/Rope.hs

```diff
@@ -81,18 +84,25 @@ import qualified Data.Text.IO as TXIO (writeFile)
 import           Data.Typeable
 import           Prelude hiding (drop)
 
+import qualified Yi.Braid as B
+import qualified Yi.Segment as S
+
 -- | Used to cache the size of the strings.
 data Size = Indices { charIndex :: {-# UNPACK #-} !Int
                       -- ^ How many characters under here?
                     , lineIndex :: Int
                       -- ^ How many lines under here?
                     } deriving (Eq, Show, Typeable)
 
+instance S.MSeg TX.Text where
```

Comment:
Hoping to find a better way to do this, open to suggestions. The end user won't really need to care about it though.